include/boost/histogram/python/register_histogram.hpp

```diff
@@ -178,82 +178,72 @@ py::class_<bh::histogram<A, S>> register_histogram(py::module &m, const char *na
             },
             "Project to a single axis or several axes on a multidiminsional histogram")
 
-        .def(make_pickle<histogram_t>())
+        .def(
+            "fill",
+            [](histogram_t &self, py::args args, py::kwargs kwargs) {
+                auto empty_weight = py::object();
+                auto weight       = optional_arg(kwargs, "weight", empty_weight);
+
+                using arrayd = py::array_t<double>;
+
+                // TODO: compute this typelist from the value types of the supported axes
+                using VArg = boost::variant2::variant<double, arrayd>;
+                auto vargs = bh::detail::make_stack_buffer<VArg>(bh::unsafe_access::axes(self));
+
+                if(args.size() != self.rank())
+                    throw std::invalid_argument("Wrong number of args");
+
+                unsigned iarg = 0;
+                for(auto arg : args) {
+                    if(py::isinstance<py::buffer>(arg) || py::hasattr(arg, "__iter__")) {
+                        auto tmp       = py::cast<arrayd>(arg);
+                        vargs.at(iarg) = tmp;
```

Comment:
It is safe to use `vargs[iarg]` here, because the line 193 already checks that iarg is within bounds.

Reply:
But it can also stay as it is, I think, this is not performance critical.

Reply:
I have a strong tendency to use `.at(...)` everywhere; I've spent lots of my early life debugging problems that turned out to be out of bounds accesses (admittedly, there are better tools now to make this slightly better). However, since this is safe due to the check above, it can be changed to `[]`. But, yes, anything that accesses Python should not be a performance critical section. The if check will only waste a few cycles, accessing Python is in the hundreds.